ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -489,58 +617,168 @@ def _useRealization(self, info, rlz):
           g.data[index, constIndex] = self._handleExplicitConstraints(newOpt, constraint)
         else:
           g.data[index, constIndex] = self._handleImplicitConstraints(newOpt, opt, constraint)
+
     offSpringFitness = self._fitnessInstance(rlz,
                                              objVar=self._objectiveVar,
                                              a=self._objCoeff,
                                              b=self._penaltyCoeff,
-                                             penalty=None,
                                              constraintFunction=g,
-                                             type=self._minMax)
+                                             constraintNum = self._numOfConst,
+                                             type = self._minMax)
+    return traj, g, objectiveVal, offSprings, offSpringFitness
 
-    self._collectOptPoint(rlz, offSpringFitness, objectiveVal,g)
-    self._resolveNewGeneration(traj, rlz, objectiveVal, offSpringFitness, g, info)
+  def singleObjSurvivorSelect(self, info, rlz, traj, offSprings, offSpringFitness, objectiveVal, g):
+      if self.counter > 1:
```

Comment:
Why are these functions in the GA? This and the next should be moved to survivorSelectors.py.